File: javascript/src/elements/missing/missing_embed.ts
import { MissingValueElementBase } from './missing_value_element_base.js';

export class MissingEmbed extends MissingValueElementBase {
    get [Symbol.toStringTag]() {
        return 'MissingEmbed';
    }
    
    toString() {
        if (this._key === null)
            return `[object MissingEmbed]`;
        
        return `[object MissingEmbed key=${this._key}]`;
    }
}

File: ruby/lib/enolib/elements/missing/missing_embed.rb
# frozen_string_literal: true

module Enolib
  class MissingEmbed < MissingValueElementBase
    def to_s
      if @key
        "#<Enolib::MissingEmbed key=#{@key}>"
      else
        '#<Enolib::MissingEmbed>'
      end
    end
  end
end
